libs/designsystem/src/lib/components/icon/icon.component.ts

```diff
@@ -54,13 +57,33 @@ export class IconComponent implements OnChanges {
     }
   }
 
-  constructor(@Optional() @Inject(ICON_SETTINGS) private iconSettings?: IconSettings) {}
+  /**
+   * iconRegistryService: Registry for custom icons.
+   * iconSettings: @deprecated Use KirbyIconRegistryService for adding custom icons.
+   */
+  constructor(
+    private iconRegistryService: KirbyIconRegistryService,
+    @Optional() @Inject(ICON_SETTINGS) private iconSettings?: IconSettings
+  ) {
+    this.combineIconSettings();
+  }
+
+  private combineIconSettings() {
+    const iconSettingsIcons: Icon[] =
+      this.iconSettings && this.iconSettings.icons
+        ? [...this.iconSettings.icons]
+        : [this.defaultIcon];
+
+    this.combinedIconSettings = {
+      icons: [...iconSettingsIcons, ...this.iconRegistryService.getCustomIcons()],
+    } as IconSettings;
+  }
 
   ngOnChanges(changes: SimpleChanges): void {
     if (changes.name && changes.name.currentValue) {
       this.icon = this.findIcon(kirbyIconSettings.icons, changes.name.currentValue);
     } else if (changes.customName && changes.customName.currentValue) {
-      this.icon = this.findIcon(this.iconSettings.icons, changes.customName.currentValue);
+      this.icon = this.findIcon(this.combinedIconSettings.icons, changes.customName.currentValue);
```

Comment:
```suggestion
      this.icon = this.findIcon(this.registeredIcons, changes.customName.currentValue);
```

Reply:
For nu synes jeg combinedIconSettings giver bedre mening da det jo er en "combined" IconSettings. Den håndtere både provided icons samt icons added med KirbyIconRegistryService

Når ICON_SETTINGS er helt deprecated kan det refactoreret